src/components/product/ProductGrid/ProductGrid.tsx

```diff
@@ -15,10 +16,34 @@ function ProductGrid({ products }: Props) {
   return (
     <div className="grid grid-cols-2 gap-2 mb-2 sm:grid-cols-4 sm:gap-7 sm:mb-7">
       {products.edges.map(({ node: product }, idx) => (
-        <ProductSummary key={`${product.id}-${idx}`} product={product} />
+        <ProductSummary
+          key={`${product.id}`}
+          product={product}
+          index={productIndex(idx)}
+        />
       ))}
     </div>
   )
 }
 
+function CurrentPage() {
+  const {
+    searchParams: { page },
+  } = useSearch()
+
+  return page
```

Comment:
This is returning 0, except if I re-render the page after clicking `show more`.
I'm not sure if that's the right behavior, what do you think, @tlgimenes?

Reply:
I get why you did this, but naming a function like this and calling a hook inside it is a React anti-pattern. Hooks must be called inside components (that by pattern always start with an uppercase letter) or inside other hooks (which must always start with `use`)

What you are doing here is naming a function starting with an uppercase letter, and although this tricks the lint, it may generate side effects and make it not work as you've intended.

To do what you want to do, the `useSearch` hook should've been called on the `ProductGrid` component and then passed as parameters to the `productIndex` function.

Reply:
Thanks for the feedback @icazevedo. I gonna change that.